### DISCUSSION TITLE
Chunking for mixed states

### DISCUSSION THREAD

**Discussion by joshuazlin** [netket/netket#1249] [Q&A] [answered]
Hello! Just wondering are there any plans for implementing chunking for mixed states?
It seems right now it is not, unless I am doing something incorrectly.

**Comment by PhilipVinc** [comment, marked as answer]
It's indeed not implemented yet, but simply because nobody needed it yet.
It should be relatively straightforward to get it done. You need to re-implement [this function](https://github.com/netket/netket/blob/05b6f4eef9b2e7a32c587ab38b41632d0808b48b/netket/vqs/mc/mc_state/expect_grad.py#L71), that computes the gradient of `LdagL` to use chunked primitives. 

The forward pass is already chunked, so you only need to chunk the backward. Though it might not be trivial to do it. 

You can find inspiration by looking at how the standard `expect_and_grad` for hermitian operators is implemented [ without chunking](https://github.com/netket/netket/blob/05b6f4eef9b2e7a32c587ab38b41632d0808b48b/netket/vqs/mc/mc_state/expect_grad.py#L42) and [with chunking](https://github.com/netket/netket/blob/05b6f4eef9b2e7a32c587ab38b41632d0808b48b/netket/vqs/mc/mc_state/expect_grad_chunked.py#L72)
